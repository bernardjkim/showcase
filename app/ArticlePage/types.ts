export interface Query {
  id: string;
}
// =============================================================================
//  ACTIONS
// =============================================================================

export const CREATE_COMMENT = 'app/ArticlePage/CREATE_COMMENT';
export const CREATE_COMMENT_SUCCESS = 'app/ArticlePage/CREATE_COMMENT_SUCCESS';
export const CREATE_COMMENT_ERROR = 'app/ArticlePage/CREATE_COMMENT_ERROR';

export const LIKE_ARTICLE = 'app/ArticlePage/LIKE_ARTICLE';
export const LIKE_ARTICLE_SUCCESS = 'app/ArticlePage/LIKE_ARTICLE_SUCCESS';
export const LIKE_ARTICLE_ERROR = 'app/ArticlePage/LIKE_ARTICLE_ERROR';

export const LOAD_ARTICLE = 'app/ArticlePage/LOAD_ARTICLE';
export const LOAD_ARTICLE_SUCCESS = 'app/ArticlePage/LOAD_ARTICLE_SUCCESS';
export const LOAD_ARTICLE_ERROR = 'app/ArticlePage/LOAD_ARTICLE_ERROR';

export const LOAD_COMMENTS = 'app/ArticlePage/LOAD_COMMENTS';
export const LOAD_COMMENTS_SUCCESS = 'app/ArticlePage/LOAD_COMMENTS_SUCCESS';
export const LOAD_COMMENTS_ERROR = 'app/ArticlePage/LOAD_COMMENTS_ERROR';

export const LOAD_LIKES = 'app/ArticlePage/LOAD_LIKES';
export const LOAD_LIKES_SUCCESS = 'app/ArticlePage/LOAD_LIKES_SUCCESS';
export const LOAD_LIKES_ERROR = 'app/ArticlePage/LOAD_LIKES_ERROR';
