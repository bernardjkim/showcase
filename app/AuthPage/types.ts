export interface SignupFormData {
  username: string;
  email: string;
  password: string;
  passwordConfirm: string;
}

export interface LoginFormData {
  email: string;
  password: string;
}

export interface AuthState {
  loading: boolean;
  // TODO: Error Def
  error: any;
}

export const CLEAR_ERRORS = 'app/AuthPage/CLEAR_ERRORS';

export const CREATE_USER = 'app/AuthPage/CREATE_USER';
export const CREATE_USER_SUCCESS = 'app/AuthPage/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'app/AuthPage/CREATE_USER_ERROR';

export const CREATE_TOKEN = 'app/App/CREATE_TOKEN';
export const CREATE_TOKEN_SUCCESS = 'app/App/CREATE_TOKEN_SUCCESS';
export const CREATE_TOKEN_ERROR = 'app/App/CREATE_TOKEN_ERROR';

export interface ClearErrorsAction {
  type: typeof CLEAR_ERRORS;
}

export interface CreateUserAction {
  type: typeof CREATE_USER;
  form: SignupFormData;
}

export interface CreateUserErrorAction {
  type: typeof CREATE_USER_ERROR;
  error: Error;
}

export interface CreateUserSuccessAction {
  type: typeof CREATE_USER_SUCCESS;
}

export interface CreateTokenAction {
  type: typeof CREATE_TOKEN;
  form: LoginFormData;
}

export interface CreateTokenErrorAction {
  type: typeof CREATE_TOKEN_ERROR;
  error: Error;
}

export interface CreateTokenSuccessAction {
  type: typeof CREATE_TOKEN_SUCCESS;
}

export type AuthActionTypes =
  | ClearErrorsAction
  | CreateTokenAction
  | CreateTokenErrorAction
  | CreateTokenSuccessAction
  | CreateUserAction
  | CreateUserErrorAction
  | CreateUserSuccessAction;
