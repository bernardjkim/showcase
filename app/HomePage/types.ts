import { Article } from 'types';
// =============================================================================
//  STATE
// =============================================================================

export interface HomeState {
  loading: boolean;
  error?: Error;
  articles: Article[];
  offset: number;
}

// =============================================================================
//  ACTIONS
// =============================================================================

export const CLEAR_STATE = 'app/HomePage/CLEAR_STATE';

export const LOAD_ARTICLES_ALL = 'app/HomePage/LOAD_ARTICLES_ALL';
export const LOAD_ARTICLES_ALL_ERROR = 'app/HomePage/LOAD_ARTICLES_ALL_ERROR';
export const LOAD_ARTICLES_ALL_SUCCESS =
  'app/HomePage/LOAD_ARTICLES_ALL_SUCCESS';

export const LOAD_NEXT = 'app/HomePage/LOAD_NEXT';
export const LOAD_NEXT_ERROR = 'app/HomePage/LOAD_NEXT_ERROR';
export const LOAD_NEXT_SUCCESS = 'app/HomePage/LOAD_NEXT_SUCCESS';

interface ClearStateAction {
  type: typeof CLEAR_STATE;
}

interface LoadArticlesAllAction {
  type: typeof LOAD_ARTICLES_ALL;
}

interface LoadArticlesAllErrorAction {
  type: typeof LOAD_ARTICLES_ALL_ERROR;
  error: Error;
}

interface LoadArticlesAllSuccessAction {
  type: typeof LOAD_ARTICLES_ALL_SUCCESS;
  articles: Article[];
}

interface LoadNextAction {
  type: typeof LOAD_NEXT;
}

interface LoadNextErrorAction {
  type: typeof LOAD_NEXT_ERROR;
  error: Error;
}

interface LoadNextSuccessAction {
  type: typeof LOAD_NEXT_SUCCESS;
  articles: Article[];
}

export type HomeActionTypes =
  | ClearStateAction
  | LoadArticlesAllAction
  | LoadArticlesAllErrorAction
  | LoadArticlesAllSuccessAction
  | LoadNextAction
  | LoadNextErrorAction
  | LoadNextSuccessAction;
