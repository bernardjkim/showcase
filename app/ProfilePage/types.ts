// =============================================================================
//  STATE
// =============================================================================

export interface ProfileState {
  loading: boolean;
  error?: Error;
}

// =============================================================================
//  ACTIONS
// =============================================================================

export type ProfileActionTypes = {};
