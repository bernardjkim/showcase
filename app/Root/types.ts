import { User } from 'types';

export interface RootState {
  loading: boolean;
  error?: Error;
  user?: User;
  validateToken: boolean;
}

// =============================================================================
//  ACTIONS
// =============================================================================

export const DELETE_TOKEN = 'app/App/DELETE_TOKEN';
export const DELETE_TOKEN_SUCCESS = 'app/App/DELETE_TOKEN_SUCCESS';
export const DELETE_TOKEN_ERROR = 'app/App/DELETE_TOKEN_ERROR';

export const LOAD_USER = 'app/App/LOAD_USER';
export const LOAD_USER_SUCCESS = 'app/App/LOAD_USER_SUCCESS';
export const LOAD_USER_ERROR = 'app/App/LOAD_USER_ERROR';

interface DeleteTokenAction {
  type: typeof DELETE_TOKEN;
}

interface DeleteTokenErrorAction {
  type: typeof DELETE_TOKEN_ERROR;
  error: Error;
}

interface DeleteTokenSuccessAction {
  type: typeof DELETE_TOKEN_SUCCESS;
}

interface LoadUserAction {
  type: typeof LOAD_USER;
}

interface LoadUserErrorAction {
  type: typeof LOAD_USER_ERROR;
  error: Error;
}

interface LoadUserSuccessAction {
  type: typeof LOAD_USER_SUCCESS;
  user: User;
}

export type RootActionTypes =
  | DeleteTokenAction
  | DeleteTokenErrorAction
  | DeleteTokenSuccessAction
  | LoadUserAction
  | LoadUserErrorAction
  | LoadUserSuccessAction;
