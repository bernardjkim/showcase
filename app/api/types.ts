import { Sort } from 'types';

export interface ArticleApi {
  create: string;
  getAll: (offset: number) => string;
  getOne: (id: string) => string;
  search: (term: string, offset: number, sort: Sort) => string;
}

export interface AuthApi {
  login: string;
  logout: string;
}

export interface CommentApi {
  create: string;
  get: (id: string) => string;
}

export interface LikeApi {
  getAll: string;
  create: string;
  getByArticle: (id: string) => string;
}

export interface UserApi {
  create: string;
  getCurrent: string;
  getOne: (id: string) => string;
}

export interface API {
  article: ArticleApi;
  auth: AuthApi;
  comment: CommentApi;
  like: LikeApi;
  user: UserApi;
}
