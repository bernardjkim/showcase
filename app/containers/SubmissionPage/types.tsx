// =============================================================================
//  STATE
// =============================================================================
export interface Form {
  screenshot: Blob;
}
export interface Error {}

export interface SubmissionState {
  loading: boolean;
  error: boolean | Error;
  submissionSuccess: boolean;
}

// =============================================================================
//  ACTIONS
// =============================================================================
export const CLEAR_STATE = 'app/SubmissionPage/CLEAR_STATE';

export const SUBMIT_FORM = 'app/SubmissionPage/SUBMIT_FORM';
export const SUBMIT_FORM_SUCCESS = 'app/SubmissionPage/SUBMIT_FORM_SUCCESS';
export const SUBMIT_FORM_ERROR = 'app/SubbmissionPage/SUBMIT_FORM_ERROR';

export interface ClearStateAction {
  type: typeof CLEAR_STATE;
}

export interface SubmitFormAction {
  type: typeof SUBMIT_FORM;
  form: Form;
}

export interface SubmitFormSuccessAction {
  type: typeof SUBMIT_FORM_SUCCESS;
}

export interface SubmitFormErrorAction {
  type: typeof SUBMIT_FORM_ERROR;
  error: Error;
}

export type SubmissionActionTypes =
  | ClearStateAction
  | SubmitFormAction
  | SubmitFormErrorAction
  | SubmitFormSuccessAction;
