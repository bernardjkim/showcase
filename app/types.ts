export interface User {
  _id: string;
  username: string;
  email: string;
  updated: Date;
}
export interface Article {
  _id: string;
  title: string;
  uri: string;
  github: string;
  description: string;
  tags: string[];
  image: string;
}

export type Sort = 'new' | 'top';
