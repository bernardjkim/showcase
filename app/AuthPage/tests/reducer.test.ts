import {
  clearErrors,
  createToken,
  createTokenError,
  createTokenSuccess,
  createUser,
  createUserError,
  createUserSuccess,
} from '../actions';
import AuthPageReducer, { initialState } from '../reducer';
import { AuthState } from '../types';

const error = new Error('Test error');
describe('AuthPageReducer', () => {
  let state: AuthState;
  beforeEach(() => {
    state = initialState;
  });

  it('handle the clear errors action', () => {
    expect(AuthPageReducer(state, clearErrors())).toMatchSnapshot();
  });

  it('handle the create token action', () => {
    const form = { email: 'email', password: 'password' };
    expect(AuthPageReducer(state, createToken(form))).toMatchSnapshot();
  });

  it('handle the create token success action', () => {
    expect(AuthPageReducer(state, createTokenSuccess())).toMatchSnapshot();
  });

  it('handle the create token error action', () => {
    expect(AuthPageReducer(state, createTokenError(error))).toMatchSnapshot();
  });

  it('handle the create user action', () => {
    const form = { username: 'username', email: 'email', password: 'password', passwordConfirm: 'password' };
    expect(AuthPageReducer(state, createUser(form))).toMatchSnapshot();
  });

  it('handle the create user success action', () => {
    expect(AuthPageReducer(state, createUserSuccess())).toMatchSnapshot();
  });

  it('handle the create user error action', () => {
    expect(AuthPageReducer(state, createUserError(error))).toMatchSnapshot();
  });
});
