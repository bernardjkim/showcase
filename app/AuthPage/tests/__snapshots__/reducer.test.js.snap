// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPageReducer handle the clear errors action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;

exports[`AuthPageReducer handle the create token action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
}
`;

exports[`AuthPageReducer handle the create token error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": Immutable.Map {
    "createToken": [Error: Test error],
  },
}
`;

exports[`AuthPageReducer handle the create token success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;

exports[`AuthPageReducer handle the create user action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
}
`;

exports[`AuthPageReducer handle the create user error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": Immutable.Map {
    "createUser": [Error: Test Error],
  },
}
`;

exports[`AuthPageReducer handle the create user success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;

exports[`AuthPageReducer returns the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;
